src/optimizer/optimizer.cpp

```diff
@@ -336,5 +339,30 @@ unique_ptr<planner::AbstractPlan> Optimizer::ChooseBestPlan(
   LOG_TRACE("Finish Choosing best plan for group %d", id);
   return plan;
 }
+
+void Optimizer::ExecuteTaskStack(
+    std::unique_ptr<OptimizerTaskStack> &task_stack, const Group *root_group,
```

Comment:
A reference to a unique_ptr seems a little bit odd to me. Could we just use a raw pointer?